### DISCUSSION TITLE
Help with redstone signal and command

### DISCUSSION THREAD

**Discussion by Undefinyx1** [cc-tweaked/CC-Tweaked#1946] [Q&A]
I want to make my speaker play an audio file after recieving a redstone input (the os.PullEvent("redstone") thingy).
Anybody know how?

**Comment by SquidDev** [comment]
Firstly, it sounds like you've figured out how to wait for a redstone input, but just in case, I normally do it by looping until [`redstone.getInput`](https://tweaked.cc/module/redstone.html#v:getInput) is true:


```lua
-- Wait for a redstone signal on the left
while not redstone.getInput("left") do
  os.pullEvent("redstone")
end
```

As to playing an audio file, there's several options. The easiest is to use [`shell.run`](https://tweaked.cc/module/shell.html#v:run) to execute the `speaker` command:

```lua
shell.run("speaker play my_audio_file.dfpwm")
```

Alternatively, you can call `playAudio` directly. The [example in the docs](https://tweaked.cc/peripheral/speaker.html#v:playAudio) is probably a good reference for that.
